src/audio/kpb.c

```diff
@@ -415,6 +415,13 @@ static int kpb_prepare(struct comp_dev *dev)
 		trace_kpb_error("kpb_prepare() error: wrong host params.");
 		return -EINVAL;
 	}
+
+	if (allocated_size && allocated_size < kpb->buffer_size) {
```

Comment:
This could be simplified and use of static allocated_size (not scalable) could be avoided. 

set kpb->buffer_size to allocated size and check in if whether < KPB_MAX_BUFFER_SIZE(sampling_width), if so, then optionally free (if not 0) and alloc to req'ed size.

Reply:
I agree, code has been updated accordingly.